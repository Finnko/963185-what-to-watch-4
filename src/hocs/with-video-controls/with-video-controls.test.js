import React from 'react';
import renderer from 'react-test-renderer';
import withVideoControls from './with-video-controls';
import FullPagePlayer from '../../components/full-page-player/full-page-player.jsx';

const props = {
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
};

const MockComponentWrapped = withVideoControls(FullPagePlayer);

describe(`Should hoc withVideoControls render correctly`, () => {
  it(`Snapshot render withVideoControls`, () => {
    const tree = renderer.create(
        <MockComponentWrapped {...props} />, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
