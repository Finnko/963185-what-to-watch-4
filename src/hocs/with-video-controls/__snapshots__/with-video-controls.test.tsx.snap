// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should hoc withVideoControls render correctly Snapshot render withVideoControls 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    onEnded={[Function]}
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
  />
  <div
    className="loader-wrapper"
    style={
      Object {
        "position": "fixed",
      }
    }
  >
    <div
      className="loader medium"
    />
  </div>
</div>
`;
